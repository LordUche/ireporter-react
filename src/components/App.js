import React, { Fragment } from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Container } from 'semantic-ui-react';
import { ToastContainer } from 'react-toastify';
import Routes from '../Routes';
import Navbar from './Navbar';
import Sidenav from './Sidenav';
import { showToast } from '../utils/helpers';
import { currentUser } from '../redux/actions/auth';

export class App extends React.Component {
  state = { sidebarOpen: false };

  async componentDidMount() {
    const { getCurrentUser } = this.props;
    await getCurrentUser();
    window.addEventListener('app-message', e =>
      showToast(e.detail.messages, e.detail.type)
    );
    window.addEventListener('offline', () =>
      showToast(['Please check your internet connectivity'], 'error')
    );
  }

  handleSidebarShow = () => this.setState({ sidebarOpen: true });

  handleSidebarHide = () => this.setState({ sidebarOpen: false });

  render() {
    const { loggedIn } = this.props;
    const { sidebarOpen } = this.state;
    return (
      <Fragment>
        <Router>
          <Sidenav visible={sidebarOpen} hideSidebar={this.handleSidebarHide}>
            <Navbar
              loggedIn={loggedIn}
              sidebarOpen={sidebarOpen}
              openSidebar={this.handleSidebarShow}
              closeSidebar={this.handleSidebarHide}
            />
            <Container className="main-content">
              <Routes />
            </Container>
          </Sidenav>
        </Router>
        <ToastContainer
          pauseOnFocusLoss={false}
          pauseOnHover={false}
          position="bottom-left"
        />
      </Fragment>
    );
  }
}

App.propTypes = {
  loggedIn: PropTypes.bool.isRequired,
  getCurrentUser: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({ loggedIn: state.auth.loggedIn });

export default connect(
  mapStateToProps,
  { getCurrentUser: currentUser }
)(App);
