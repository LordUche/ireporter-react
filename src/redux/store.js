import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { logger } from 'redux-logger';
import thunk from 'redux-thunk';
import rootReducer from './reducers';

const middleware = [thunk];

process.env.NODE_ENV === `development` && middleware.push(logger);

const initialState = { auth: { loggedIn: false, loading: false } };
const store = createStore(
  rootReducer,
  initialState,
  composeWithDevTools(applyMiddleware(...middleware))
);

export default store;
