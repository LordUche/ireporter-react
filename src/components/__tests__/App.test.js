import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Provider } from 'react-redux';
import App from '../App';
import store from '../../redux/store';

describe('<App />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });

  it('should match snapshot', () => {
    const tree = toJSON(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
