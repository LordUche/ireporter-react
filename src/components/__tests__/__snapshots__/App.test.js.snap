// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ConnectFunction>
    <App
      getCurrentUser={[Function]}
      loggedIn={false}
    >
      <BrowserRouter>
        <Router
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
        >
          <ConnectFunction
            hideSidebar={[Function]}
            visible={false}
          >
            <Sidenav
              hideSidebar={[Function]}
              loggedIn={false}
              logout={[Function]}
              visible={false}
            >
              <SidebarPushable
                as={[Function]}
              >
                <Segment
                  className="pushable"
                >
                  <div
                    className="ui segment pushable"
                  >
                    <Sidebar
                      animation="uncover"
                      as={[Function]}
                      direction="right"
                      icon="labeled"
                      onHide={[Function]}
                      vertical={true}
                      visible={false}
                      width="thin"
                    >
                      <Ref
                        innerRef={
                          Object {
                            "current": <div
                              class="ui vertical labeled icon ui uncover right thin sidebar menu"
                            >
                              <a
                                class="item"
                                href="/"
                              >
                                <i
                                  aria-hidden="true"
                                  class="home icon"
                                />
                                Home
                              </a>
                              <a
                                class="item sidebar__login-button"
                                href="/login"
                              >
                                <i
                                  aria-hidden="true"
                                  class="sign in icon"
                                />
                                Log In
                              </a>
                              <a
                                class="item sidebar__signup-button"
                                href="/signup"
                              >
                                <i
                                  aria-hidden="true"
                                  class="signup icon"
                                />
                                Sign Up
                              </a>
                            </div>,
                          }
                        }
                      >
                        <RefFindNode
                          innerRef={
                            Object {
                              "current": <div
                                class="ui vertical labeled icon ui uncover right thin sidebar menu"
                              >
                                <a
                                  class="item"
                                  href="/"
                                >
                                  <i
                                    aria-hidden="true"
                                    class="home icon"
                                  />
                                  Home
                                </a>
                                <a
                                  class="item sidebar__login-button"
                                  href="/login"
                                >
                                  <i
                                    aria-hidden="true"
                                    class="sign in icon"
                                  />
                                  Log In
                                </a>
                                <a
                                  class="item sidebar__signup-button"
                                  href="/signup"
                                >
                                  <i
                                    aria-hidden="true"
                                    class="signup icon"
                                  />
                                  Sign Up
                                </a>
                              </div>,
                            }
                          }
                        >
                          <Menu
                            className="ui uncover right thin sidebar"
                            icon="labeled"
                            vertical={true}
                          >
                            <div
                              className="ui vertical labeled icon ui uncover right thin sidebar menu"
                            >
                              <MenuItem
                                as={[Function]}
                                onClick={[Function]}
                                to="/"
                              >
                                <Link
                                  className="item"
                                  onClick={[Function]}
                                  to="/"
                                >
                                  <a
                                    className="item"
                                    href="/"
                                    onClick={[Function]}
                                  >
                                    <Icon
                                      as="i"
                                      name="home"
                                    >
                                      <i
                                        aria-hidden="true"
                                        className="home icon"
                                        onClick={[Function]}
                                      />
                                    </Icon>
                                    Home
                                  </a>
                                </Link>
                              </MenuItem>
                              <MenuItem
                                as={[Function]}
                                className="sidebar__login-button"
                                onClick={[Function]}
                                to="/login"
                              >
                                <Link
                                  className="item sidebar__login-button"
                                  onClick={[Function]}
                                  to="/login"
                                >
                                  <a
                                    className="item sidebar__login-button"
                                    href="/login"
                                    onClick={[Function]}
                                  >
                                    <Icon
                                      as="i"
                                      name="sign in"
                                    >
                                      <i
                                        aria-hidden="true"
                                        className="sign in icon"
                                        onClick={[Function]}
                                      />
                                    </Icon>
                                    Log In
                                  </a>
                                </Link>
                              </MenuItem>
                              <MenuItem
                                as={[Function]}
                                className="sidebar__signup-button"
                                onClick={[Function]}
                                to="/signup"
                              >
                                <Link
                                  className="item sidebar__signup-button"
                                  onClick={[Function]}
                                  to="/signup"
                                >
                                  <a
                                    className="item sidebar__signup-button"
                                    href="/signup"
                                    onClick={[Function]}
                                  >
                                    <Icon
                                      as="i"
                                      name="signup"
                                    >
                                      <i
                                        aria-hidden="true"
                                        className="signup icon"
                                        onClick={[Function]}
                                      />
                                    </Icon>
                                    Sign Up
                                  </a>
                                </Link>
                              </MenuItem>
                            </div>
                          </Menu>
                        </RefFindNode>
                      </Ref>
                    </Sidebar>
                    <SidebarPusher>
                      <div
                        className="pusher"
                      >
                        <main>
                          <ConnectFunction
                            closeSidebar={[Function]}
                            loggedIn={false}
                            openSidebar={[Function]}
                            sidebarOpen={false}
                          >
                            <Navbar
                              closeSidebar={[Function]}
                              loggedIn={false}
                              logout={[Function]}
                              openSidebar={[Function]}
                              sidebarOpen={false}
                              user={null}
                            >
                              <nav
                                className="navbar"
                              >
                                <Segment
                                  inverted={true}
                                >
                                  <div
                                    className="ui inverted segment"
                                  >
                                    <Menu
                                      inverted={true}
                                      secondary={true}
                                    >
                                      <div
                                        className="ui inverted secondary menu"
                                      >
                                        <Container>
                                          <div
                                            className="ui container"
                                          >
                                            <MenuItem
                                              as={[Function]}
                                              className="navbar__brand"
                                              icon={true}
                                              to="/"
                                            >
                                              <Link
                                                className="icon item navbar__brand"
                                                onClick={[Function]}
                                                to="/"
                                              >
                                                <a
                                                  className="icon item navbar__brand"
                                                  href="/"
                                                  onClick={[Function]}
                                                >
                                                  <Image
                                                    as="img"
                                                    className="navbar__brand--icon"
                                                    size="mini"
                                                    spaced="right"
                                                    src="test-file-stub"
                                                    ui={true}
                                                  >
                                                    <img
                                                      className="ui mini right spaced image navbar__brand--icon"
                                                      src="test-file-stub"
                                                    />
                                                  </Image>
                                                  <Header
                                                    className="navbar__brand--text"
                                                  >
                                                    <div
                                                      className="ui header navbar__brand--text"
                                                    >
                                                      iREPORTER
                                                    </div>
                                                  </Header>
                                                </a>
                                              </Link>
                                            </MenuItem>
                                            <Responsive
                                              as={[Function]}
                                              getWidth={[Function]}
                                              minWidth={769}
                                              position="right"
                                            >
                                              <MenuMenu
                                                position="right"
                                              >
                                                <div
                                                  className="right menu"
                                                >
                                                  <MenuItem
                                                    as={[Function]}
                                                    name="sign up"
                                                    to="/signup"
                                                  >
                                                    <Link
                                                      className="item"
                                                      onClick={[Function]}
                                                      to="/signup"
                                                    >
                                                      <a
                                                        className="item"
                                                        href="/signup"
                                                        onClick={[Function]}
                                                      >
                                                        Sign Up
                                                      </a>
                                                    </Link>
                                                  </MenuItem>
                                                  <MenuItem
                                                    as={[Function]}
                                                    name="log in"
                                                    to="/login"
                                                  >
                                                    <Link
                                                      className="item"
                                                      onClick={[Function]}
                                                      to="/login"
                                                    >
                                                      <a
                                                        className="item"
                                                        href="/login"
                                                        onClick={[Function]}
                                                      >
                                                        Log In
                                                      </a>
                                                    </Link>
                                                  </MenuItem>
                                                </div>
                                              </MenuMenu>
                                            </Responsive>
                                            <Responsive
                                              as={[Function]}
                                              getWidth={[Function]}
                                              maxWidth={768}
                                              position="right"
                                            />
                                          </div>
                                        </Container>
                                      </div>
                                    </Menu>
                                  </div>
                                </Segment>
                              </nav>
                            </Navbar>
                          </ConnectFunction>
                          <Container
                            className="main-content"
                          >
                            <div
                              className="ui container main-content"
                            >
                              <Routes>
                                <Switch>
                                  <Route
                                    component={[Function]}
                                    computedMatch={
                                      Object {
                                        "isExact": true,
                                        "params": Object {},
                                        "path": "/",
                                        "url": "/",
                                      }
                                    }
                                    exact={true}
                                    location={
                                      Object {
                                        "hash": "",
                                        "pathname": "/",
                                        "search": "",
                                        "state": undefined,
                                      }
                                    }
                                    path="/"
                                  >
                                    <Landing
                                      history={
                                        Object {
                                          "action": "POP",
                                          "block": [Function],
                                          "createHref": [Function],
                                          "go": [Function],
                                          "goBack": [Function],
                                          "goForward": [Function],
                                          "length": 1,
                                          "listen": [Function],
                                          "location": Object {
                                            "hash": "",
                                            "pathname": "/",
                                            "search": "",
                                            "state": undefined,
                                          },
                                          "push": [Function],
                                          "replace": [Function],
                                        }
                                      }
                                      location={
                                        Object {
                                          "hash": "",
                                          "pathname": "/",
                                          "search": "",
                                          "state": undefined,
                                        }
                                      }
                                      match={
                                        Object {
                                          "isExact": true,
                                          "params": Object {},
                                          "path": "/",
                                          "url": "/",
                                        }
                                      }
                                    >
                                      <div
                                        className="landing"
                                      >
                                        <Header
                                          as="h1"
                                          className="landing-header"
                                        >
                                          <h1
                                            className="ui header landing-header"
                                          >
                                            See something? 
                                            <br />
                                             Report it!
                                          </h1>
                                        </Header>
                                        <HeaderSubheader
                                          className="landing__subheader"
                                        >
                                          <div
                                            className="sub header landing__subheader"
                                          >
                                            <span>
                                              Say 
                                              <span
                                                className="color-red"
                                              >
                                                NO
                                              </span>
                                               to corruption
                                            </span>
                                            <br />
                                            <span>
                                              Be the voice in your community
                                            </span>
                                          </div>
                                        </HeaderSubheader>
                                        <Button
                                          as={[Function]}
                                          color="red"
                                          content="GET STARTED"
                                          to="/signup"
                                        >
                                          <Ref
                                            innerRef={
                                              Object {
                                                "current": <a
                                                  class="ui red button"
                                                  href="/signup"
                                                  role="button"
                                                >
                                                  GET STARTED
                                                </a>,
                                              }
                                            }
                                          >
                                            <RefFindNode
                                              innerRef={
                                                Object {
                                                  "current": <a
                                                    class="ui red button"
                                                    href="/signup"
                                                    role="button"
                                                  >
                                                    GET STARTED
                                                  </a>,
                                                }
                                              }
                                            >
                                              <Link
                                                className="ui red button"
                                                onClick={[Function]}
                                                role="button"
                                                to="/signup"
                                              >
                                                <a
                                                  className="ui red button"
                                                  href="/signup"
                                                  onClick={[Function]}
                                                  role="button"
                                                >
                                                  GET STARTED
                                                </a>
                                              </Link>
                                            </RefFindNode>
                                          </Ref>
                                        </Button>
                                      </div>
                                    </Landing>
                                  </Route>
                                </Switch>
                              </Routes>
                            </div>
                          </Container>
                        </main>
                      </div>
                    </SidebarPusher>
                  </div>
                </Segment>
              </SidebarPushable>
            </Sidenav>
          </ConnectFunction>
        </Router>
      </BrowserRouter>
      <ToastContainer
        autoClose={5000}
        bodyClassName={null}
        className={null}
        closeButton={
          <CloseButton
            ariaLabel="close"
          />
        }
        closeOnClick={true}
        draggable={true}
        draggablePercent={80}
        hideProgressBar={false}
        newestOnTop={false}
        pauseOnFocusLoss={true}
        pauseOnHover={true}
        position="top-right"
        progressClassName={null}
        progressStyle={null}
        rtl={false}
        style={null}
        toastClassName={null}
        transition={[Function]}
      >
        <div
          className="Toastify"
        />
      </ToastContainer>
    </App>
  </ConnectFunction>
</Provider>
`;
